src/QsCompiler/TextProcessor/SyntaxBuilder.fs

```diff
@@ -345,17 +345,25 @@ let internal symbolNameLike errCode =
                      preCheckStart = isSymbolStart,
                      preCheckContinue = isSymbolContinuation) |> identifier
         getPosition .>>. id .>>. getPosition |>> fun ((p1, name), p2) -> name, (p1,p2)
-    let whenValid (name : string, range) = 
-        let isReserved = name.StartsWith "__" && name.EndsWith "__" || InternalUse.CsKeywords.Contains name
+    let whenValid ((name : string, range), isBeforeDot) =
+        // REL0920: 
+        // The warning for reservedUnderscorePattern should be replaced with an error in the future, 
+        // and the first half of isReserved should be removed.
+        let reservedUnderscorePattern = name.Contains "__" || (isBeforeDot && name.EndsWith "_")
```

Comment:
I know this code is temporary - but maybe we can say `futureReservedUnderscores` in contrast with `isReserved` (which also contains a reserved underscore pattern)?

Reply:
Sure.